Demo/HelloWorld/Client/src/Program.cs

```diff
@@ -0,0 +1,212 @@
+using Google.Protobuf.WellKnownTypes;
+
+using static System.Console;
+
+
+namespace ArmoniK.Samples.HelloWorld.Client;
+
+internal class Program
+{
+  public static async Task<int> Main(string[] args)
+  {
+    // Define the options for the application with their description and default value
+    var endpoint = new Option<string>(name: "--endpoint",
+                                      description: "Endpoint for the connection to ArmoniK control plane.",
+                                      getDefaultValue: () => "http://localhost:5001");
+    var partition = new Option<string>(name: "--partition",
+                                       description: "Name of the partition to which submit tasks.",
+                                       getDefaultValue: () => "default");
+    var input = new Option<string>(name: "--input",
+                                   description: "Input string for the tasks.",
+                                   getDefaultValue: () => "Hello");
+    var nTasks = new Option<int>(name: "--nTasks",
+                                 description: "Number of tasks to send to ArmoniK.",
+                                 getDefaultValue: () => 1);
+    var nResultsPerTask = new Option<int>(name: "--nResultsPerTask",
+                                          description: "Number of results per tasks.",
+                                          getDefaultValue: () => 1);
+
+    // Describe the application and its purpose
+    var rootCommand =
+      new
+        RootCommand($"Hello World demo for ArmoniK.\nIt sends <{nTasks.Name}> to ArmoniK with <{nResultsPerTask.Name}> per task in the given partition <{partition.Name}>. Each task receive <{input.Name}> as input string and, for each result that will be returned by the task, append the word 'World' and the resultID to the <{input.Name}>. Then, the client retrieves and prints the results for each task.\nArmoniK endpoint location is provided through <{endpoint.Name}>");
+
+    // Add the options to the parser
+    rootCommand.AddOption(endpoint);
+    rootCommand.AddOption(partition);
+    rootCommand.AddOption(input);
+    rootCommand.AddOption(nTasks);
+    rootCommand.AddOption(nResultsPerTask);
+
+    // Configure the handler to call the function that will do the work
+    rootCommand.SetHandler(async (e,
+                                  p,
+                                  s,
+                                  tasks,
+                                  resultsPerTask) =>
+                           {
+                             await Run(e,
+                                       p,
+                                       s,
+                                       tasks,
+                                       resultsPerTask);
+                           },
+                           endpoint,
+                           partition,
+                           input,
+                           nTasks,
+                           nResultsPerTask);
```

Comment:
Isn't there a simpler way to write this? This looks like heavy boilerplate that should not exist. Maybe at least:
```suggestion
    rootCommand.SetHandler(Run,
                           endpoint,
                           partition,
                           input,
                           nTasks,
                           nResultsPerTask);
```